File: app/src/test/java/org/schulcloud/mobile/data/model/EventTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;
import org.schulcloud.mobile.data.model.jsonApi.Included;

import io.realm.RealmList;

import static org.junit.Assert.assertEquals;

public class EventTest {
    private static final String _ID = "ID";
    private static final String TYPE = "TEST";
    private static final String TITLE = "TEST_PATH";
    private static final Boolean ALLDAY = false;
    private static final String START = "IMAGE";
    private static final String END = "TEST_THUMBNAIL";
    private static final String SUMMARY = "IMAGE";
    private static final String LOCATION = "TEST_THUMBNAIL";
    private static final RealmList<Included> INCLUDED = new RealmList<>();
    private static final String XSCCOURSEID = "COURSE";
    private static final String XSCCOURSETIMEID = "COURSEID";

    private Event event;

    @Before
    public void setUp() {
        event = createNewEvent();
    }

    @Test
    public void testGetProperties() {
        assertEquals(event._id, _ID);
        assertEquals(event.type, TYPE);
        assertEquals(event.title, TITLE);
        assertEquals(event.allDay, ALLDAY);
        assertEquals(event.start, START);
        assertEquals(event.end, END);
        assertEquals(event.summary, SUMMARY);
        assertEquals(event.location, LOCATION);
        assertEquals(event.included, INCLUDED);
        assertEquals(event.xScCourseId, XSCCOURSEID);
        assertEquals(event.xScCourseTimeId, XSCCOURSETIMEID);
    }

    public static Event createNewEvent() {
        Event event = new Event();
        event._id = _ID;
        event.type = TYPE;
        event.title = TITLE;
        event.allDay = ALLDAY;
        event.start = START;
        event.end = END;
        event.summary = SUMMARY;
        event.location = LOCATION;
        event.included = INCLUDED;
        event.xScCourseId = XSCCOURSEID;
        event.xScCourseTimeId = XSCCOURSETIMEID;

        return event;
    }
}

File: app/src/test/java/org/schulcloud/mobile/models/event/EventTest.kt
package org.schulcloud.mobile.models.event

import org.junit.Before

import org.junit.Assert.*
import io.realm.RealmList
import org.junit.Test

class EventTest {
    private companion object {
        const val ID = "id"
        const val TYPE = "type"
        const val TITLE = "title"
        const val ALLDAY = false
        const val START = 1L
        const val END = 2L
        const val SUMMARY = "summary"
        const val LOCATION = "location"
        const val DESCRIPTION = "description"
        val INCLUDED = RealmList<Included>()
        const val COURSEID = "course"
        const val DURATION = 1L

    }

    private val newEvent: Event
        get() = Event().apply {
            id = ID
            type = TYPE
            title = TITLE
            allDay = ALLDAY
            start = START
            end = END
            summary = SUMMARY
            location = LOCATION
            description = DESCRIPTION
            included = INCLUDED
            courseId = COURSEID
        }

    private lateinit var event: Event

    @Before
    fun setUp() {
        event = newEvent
    }

    @Test
    fun testGetProperties() {
        assertEquals(event.id, ID)
        assertEquals(event.type, TYPE)
        assertEquals(event.title, TITLE)
        assertEquals(event.allDay, ALLDAY)
        assertEquals(event.start, START)
        assertEquals(event.end, END)
        assertEquals(event.summary, SUMMARY)
        assertEquals(event.location, LOCATION)
        assertEquals(event.included, INCLUDED)
        assertEquals(event.courseId, COURSEID)
        assertEquals(event.duration, DURATION)
    }
    // TODO: test nextStart too
}
